Create occupancy grid in arbitrary frame

embasm opened the pull request:
To create occupancy grid in arbitrary frame, modify to get transform
matrix from tf and give it to occupancy grid rendering.

embasm committed:
Create occupancy grid in arbitary frame

To create occupancy grid in arbitrary frame, modify to get transform
matrix from tf and give it to occupancy grid rendering.

mohammedari commented on opengl_ros_lib/src/depth_image_projector.cpp:
```diff
@@ -68,6 +68,10 @@ struct DepthImageProjector::Impl
+    std::array<float, 16> depthToMap_ = {0.0f, -1.0f,  0.0f, 0.0f,
+                                         0.0f,  0.0f, -1.0f, 0.0f,
+                                         1.0f,  0.0f,  0.0f, 0.0f,
+                                         0.0f,  0.0f,  0.0f, 1.0f};
```
This array initialization in struct causes error when compiled with gcc 5.3, which is the newest version distributed with Ubuntu 16.04. Although this is [a bug of gcc](https://stackoverrun.com/ja/q/9068088), could you add another brace for work around to be successfully compiled with Ubuntu 16.04 environment?

mohammedari commented on opengl_ros_lib/src/depth_image_projector.cpp:
```diff
@@ -196,6 +201,12 @@ void DepthImageProjector::Impl::project(cv::Mat& dest, const cv::Mat& color, con
+        std::array<float, 4> normalized_camera_pos = {
+          depthToMap_[12] / gridMapResolution_ / gridMapWidth_ * 2,
+          depthToMap_[13] / gridMapResolution_ / gridMapHeight_ * 2,
+          depthToMap_[14] / gridMapLayerHeight_ / 2,
```
This line should be `depthToMap_[14] / gridMapLayerHeight_ * 2` since -height/2 to height/2 is mapped to -1 to 1 in NDC space.

mohammedari reviewed (commented):
Thank you for opening a PR! Could you look into lines I commented below?

embasm committed:
Fix the origin of a camera for image projection

embasm committed:
Fix initialization of std::array for gcc-5.3

This commit adds another braces to avoid the bug of gcc-5.3 which
causes a compilation error when std::array is initialized with single
braces.

embasm commented:
@mohammedari
As per your comments, I fixed the code.
Please confirm whether your concern is resolved.

mohammedari commented:
I confirmed build passed on Ubuntu 16.04. Thank you for the fix! LGTM!
